### PR TITLE
fix: id003 initialization improvements

### PR DESCRIPTION
**Pull request by RafaelTaranto**

### CODE REVIEW

**Comment by naconner** [comment]
[id003 initialization improvements](https://trello.com/c/Fou4uXnt/136-id003-initialization-improvements)

**Comment by joshmh** [comment]
What happens if we miss the ENQ response? Since we're not polling, we'll never check the status. Also, it might still be possible to send two commands at the same time, although that does become more unlikely.

It might be simpler to instead have a function that controls all sends and makes sure they are always properly serialized. It could also drop status requests if not enough time has elapsed since the previous command was sent.

File: lib/id003/id003.js [hunk 1]
```diff
@@ -238,16 +240,22 @@ Id003.prototype._denominations = function _denominations () {
...
-    this._startPolling()
+    // we only want polling before going to interrupt mode
+    if (!this.disablePolling) {
+      this._startPolling()
+    }
   }, POLLING_INTERVAL)
 }
 
 Id003.prototype._startPolling = function () {
+  clearInterval(this.pollingInterval)
```
Comment (joshmh) [inline lib/id003/id003.js:258]: Might as well use ``_stopPolling()`` here.

**Review by joshmh — CHANGES_REQUESTED** [review]

File: lib/id003/id003.js [hunk 2]
```diff
@@ -238,16 +240,22 @@ Id003.prototype._denominations = function _denominations () {
...
+  if (this.disablePolling) return this.rs232.send(command, data)
+
   this._stopPolling()
   // timeout because last poll could have been at now minus 1ms
   // potentially sending a new request before getting the response
   setTimeout(() => {
     this.rs232.send(command, data)
-    this._startPolling()
+    // we only want polling before going to interrupt mode
+    if (!this.disablePolling) {
```
[the remark this answers is not in the document]
Comment (RafaelTaranto) [reply lib/id003/id003.js:251]: good point.

[the remark this answers is not in the document]
Comment (RafaelTaranto) [reply lib/id003/id003.js]: actually since this is a arrow function this can be updated by the time we get here.

**Pull request merged**